packages/@lwc/template-compiler/src/parser/parser.ts

```diff
@@ -54,6 +54,30 @@ export default class ParserCtx {
         return this.source.slice(start, end);
     }
 
+    *ancestorGenerator(element?: IRElement) {
+        const ancestors = element ? [...this.parentStack, element] : this.parentStack;
```

Comment:
I'm not sure if `[...this.parentStack, element]` will be a performance issue since it creates a copy of parentStack each time the function is called with an argument.  I wrote a version without the copy as well:

```
*ancestorGenerator(element?: IRElement) {
    let size = this.parentStack.length;
    let current = element || this.parentStack[--size];
    for (let index = size; index >= 0 && current; index--) {
        yield { current, index };
        current = this.parentStack[index-1];
    }
}
```

Reply:
I am fine either way. I am not concerned by the performance overhead of the array spread.